### PR TITLE
Feature/http client risky cleanup

### PR DESCRIPTION
**Pull request by denis-yuen**
Random cleanup stuff that was in previous PR but seemed risky at the time
https://github.com/dockstore/dockstore/pull/4120

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java [hunk 1]
```diff
@@ -240,12 +237,12 @@ protected Tool unsupportedOperation() throws CustomWebApplicationException {
...
             List<Tag> toolTags = getTags(tool);
             final SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory
-                .createSourceCodeRepo(tool.getGitUrl(), client, bitbucketToken == null ? null : bitbucketToken.getContent(),
+                .createSourceCodeRepo(tool.getGitUrl(), bitbucketToken == null ? null : bitbucketToken.getContent(),
                     gitlabToken == null ? null : gitlabToken.getContent(), githubToken);
             updateTags(toolTags, tool, sourceCodeRepo, tagDAO, fileDAO, toolDAO, fileFormatDAO, eventDAO, user);
         }
 
-        return newDBTools;
+        return;
```
Comment (GFJHogue) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java:245]: Codacy calls this out (void return)

**Review by GFJHogue — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java [hunk 2]
```diff
@@ -44,13 +43,13 @@ private SourceCodeRepoFactory() {
 
     public static SourceCodeRepoInterface createGitHubAppRepo(String token) {
         // The gitUsername doesn't seem to matter
-        return new GitHubSourceCodeRepo("dockstore", token, "JWT");
+        return new GitHubSourceCodeRepo("JWT", token);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java:46]: Won't there be the same issue risk of GitHub apps clobbering each others' caches? May not be a enough of issue for our tests though.

**Review by coverbeck — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java (hunk 2 above)
Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java:46]: Not sure, but it also should not change. The JTW tokens are supposed to expire every ten minutes anyway so cachiing is a bit futile except for rapid requests for the same content

Comment (coverbeck) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java:46]: There all the getBranches/getTags calls, but those probably don't add up to much.

**Pull request merged**